scripts/core/ArticlesListByQuery.tsx

```diff
@@ -0,0 +1,205 @@
+/* eslint-disable react/no-multi-comp */
+
+import React from 'react';
+import {ArticlesListV2} from './ArticlesListV2';
+import {IRestApiResponse, IArticle} from 'superdesk-api';
+import {httpRequestJsonLocal} from './helpers/network';
+import {flatMap, once} from 'lodash';
+import ng from 'core/services/ng';
+import {ISuperdeskQuery, toElasticQuery, getQueryFieldsRecursive} from './query-formatting';
+import {Loader} from 'superdesk-ui-framework/react/components/Loader';
+import {IMultiSelectNew} from 'apps/search/components/ItemList';
+
+interface IProps {
+    query: ISuperdeskQuery;
+    onItemClick(item: IArticle): void;
+    onItemDoubleClick(item: IArticle): void;
+    header?(itemCount: number): JSX.Element;
+    padding?: string;
+    multiSelect?: IMultiSelectNew;
+}
+
+interface IPropsInner extends IProps {
+    beforeUnmount(): void;
+    afterInitialized(): void;
+}
+
+interface IState {
+    itemCount: number | 'loading';
+}
+
+class ArticlesListByQueryComponent extends React.PureComponent<IPropsInner, IState> {
+    markAsInitializedOnce: () => void;
+
+    constructor(props: IPropsInner) {
+        super(props);
+
+        this.state = {
+            itemCount: 'loading',
+        };
+
+        this.loadItems = this.loadItems.bind(this);
+        this.markAsInitializedOnce = once(() => {
+            this.props.afterInitialized();
+        });
+    }
+    loadItems(from, to): Promise<IRestApiResponse<any>> {
+        const pageSize = to - from;
+
+        const withPagination = {
+            ...this.props.query,
+            page: Math.floor(from / pageSize),
+            max_results: pageSize,
+        };
+
+        const query = toElasticQuery(withPagination);
+
+        return httpRequestJsonLocal<IRestApiResponse<IArticle>>({
+            method: 'GET',
+            path: '/search',
+            urlParams: {
+                aggregations: 0,
+                es_highlight: 1,
+                projections: JSON.stringify(ng.get('search').getProjectedFields()),
+                source: JSON.stringify(query),
+            },
+        }).then((res) => {
+            return new Promise((resolve) => {
+                // update item count
+
+                this.setState({itemCount: res._meta.total}, () => {
+                    resolve(res);
+                });
+            });
+        });
+    }
+    componentDidMount() {
+        this.loadItems(0, 1).then((res) => {
+            this.setState({
+                itemCount: res._meta.total,
```

Comment:
It's only about getting the count. It was required for a virtual scrolling library so it can compute the correct height of the scroll-box. I ended up not using that library due to performance issues of our list item component, but I'll be doing more work on it so I'll keep that logic for now.

I think I've tried to call it with `0, 0` but server threw an error.